functions/firestore/index.js

```diff
@@ -22,16 +22,27 @@ module.exports = runWith({ secrets: ['SENDGRID_API_KEY'] })
   .firestore.document(
     `${config().db.firestore_nft_mint_subscriptions_root}/{id}`
   )
-  .onCreate(async snapshot => {
-    const subscriptionId = snapshot.id;
-    const { subscriberEmail, subscribedWallet } = snapshot.data();
+  .onWrite(async change => {
+    if (!change.after || !change.after.data()) {
+      return Promise.resolve();
+    }
+
+    const {
+      subscriberEmail,
+      subscribedWallet,
+      isVerified,
+    } = change.after.data();
+    if (isVerified) {
+      return Promise.resolve();
```

Comment:
if these `return Promise.resolve()` is for type consistency, I think awaiting `sendEmail` below before returning, and switch to return primitive types across this function is simpler